### ISSUE TITLE
Linking against `libpanic_unwind` failed on macos due to `lib.rmeta` being unrecognizable by `ld`

### ISSUE THREAD

**Issue by NobodyXu** [rust-lang/rust#107252]
The [`CI`](https://github.com/cargo-bins/jobslot/actions/runs/3984951742/jobs/6831744278#step:6:68) of `jobslot` failed when linking against `libpanic_unwind` on macos:

```
  = note: ld: in /Users/runner/.rustup/toolchains/1.52-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-2fe2b8cd4fa63c34.rlib(lib.rmeta), archive member 'lib.rmeta' with length 35552 is not mach-o or llvm bitcode file '/Users/runner/.rustup/toolchains/1.52-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-2fe2b8cd4fa63c34.rlib'
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

### Meta

`cargo` toolchain installed via `rustup`:
```
1.52-x86_64-apple-darwin
```

GHA runner info:

```
Current runner version: '2.300.2'
Operating System
  macOS
  12.6.2
  21G320
Runner Image
  Image: macos-12
  Version: 20230117.3
  Included Software: https://github.com/actions/runner-images/blob/macOS-12/20230117.3/images/macos/macos-12-Readme.md
  Image Release: https://github.com/actions/runner-images/releases/tag/macOS-12%2F20230117.3
Runner Image Provisioner
  2.0.98.1
```

**Comment by NobodyXu** [comment]
This error seems to also happen in https://github.com/sagebind/castaway/pull/11

**Comment by bjorn3** [comment]
Duplicate of https://github.com/rust-lang/rust/issues/105167

tl;dr: XCode 14 broke rustc 1.53 and older. You have to update to at least rustc 1.54 or downgrade to XCode 13 or older.

**Comment by NobodyXu** [comment]
Thanks, I will try lld to see if that works.

**Comment by NobodyXu** [comment]
@bjorn3 Thanks, I've already bumped it to 1.56 as required by windows-sys 0.48
